src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -12,29 +12,47 @@ import {
     LogStreamRegistry,
     filterLogEventsFromUriComponents,
 } from '../registry/logStreamRegistry'
-import { createQuickPick, DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+import { DataQuickPickItem } from '../../shared/ui/pickerPrompter'
 import { Wizard } from '../../shared/wizards/wizard'
-import { createInputBox } from '../../shared/ui/inputPrompter'
+import { createInputBox, InputBoxPrompter } from '../../shared/ui/inputPrompter'
 import { createURIFromArgs } from '../cloudWatchLogsUtils'
 import { DefaultCloudWatchLogsClient } from '../../shared/clients/cloudWatchLogsClient'
 import { CloudWatchLogs } from 'aws-sdk'
+import { RegionSubmenu, RegionSubmenuResponse } from '../../shared/ui/common/regionSubmenu'
+import { integer } from 'aws-sdk/clients/cloudfront'
+import { Prompter, PromptResult } from '../../shared/ui/prompter'
+import { createQuickPick, QuickPickPrompter } from '../../shared/ui/pickerPrompter'
+import { ItemLoadTypes } from '../../shared/ui/pickerPrompter'
+import { isValidResponse } from '../../shared/wizards/wizard'
+import { StepEstimator } from '../../shared/wizards/wizard'
 import { highlightDocument } from '../document/logStreamDocumentProvider'
+import { CancellationError } from '../../shared/utilities/timeoutUtils'
+import { localize } from 'vscode-nls'
+import { getLogger } from '../../shared/logger'
 
 export async function searchLogGroup(registry: LogStreamRegistry): Promise<void> {
     let result: telemetry.Result = 'Succeeded'
-    const regionCode = 'us-west-2'
-    const client = new DefaultCloudWatchLogsClient(regionCode)
-    const logGroups = await logGroupsToArray(client.describeLogGroups())
-    const response = await new SearchLogGroupWizard(logGroups).run()
+    const response = await new SearchLogGroupWizard().run()
     if (response) {
         const logGroupInfo: CloudWatchLogsGroupInfo = {
-            groupName: response.logGroup,
-            regionName: regionCode,
+            groupName: response.submenuResponse.data,
+            regionName: response.submenuResponse.region,
         }
+        let parameters: CloudWatchLogsParameters
 
-        const parameters: CloudWatchLogsParameters = {
-            limit: registry.configuration.get('limit', 10000),
-            filterPattern: response.filterPattern,
+        if (response.timeRange.start === response.timeRange.end) {
+            // this means no time filter.
+            parameters = {
+                limit: registry.configuration.get('limit', 10000),
```

Comment:
Does it make sense to extract:
```javascript
registry.configuration.get('limit', 10000)
```
outside the if statement since it's being used in both (just to make it easier if the 10000 limit changes and its always going to be the same in both)